packages/server/src/api/controllers/row/internalSearch.js

```diff
@@ -304,6 +304,26 @@ const recursiveSearch = async (query, params) => {
   if (rows.length > params.limit - 200) {
     pageSize = params.limit - rows.length
   }
+
+  /*
+  If a null value is passed to a filter expression, we currently return 
+  every row, this can and has resulted in data loss via automation loops etc.
+  So here we are checking if any null values exist in the filters, and if they do
+  just return an empty rows array.
+  */
+  let nullValuesExist = false
+  Object.keys(query).forEach(key => {
```

Comment:
Since this loops over every operator, this means that if you wanted to do "not equals empty string", or "not equals null" that those are now impossible too since you'll just get back no results. Some operators don't have values either (like "is empty" or "is not empty" - but I think since those should be `undefined` and you're doing an explicit `=== null` that hopefully those still work. Could you double check that to make sure?

Reply:
Yea I see your point on the first one, I don't really see another way to handle this other than potentially only doing the null check on specific operators, maybe `string` `fuzzy` `equal`?

Reply:
It's a tough one because I see what you're saying that when doing things like looping in automations with a delete row action or something, that you want to be conservative. But 95% of the time you're just pulling data, for which you would want to do the opposite and pull back more data than necessary rather than less. What was the original issue that led to data loss? I'm wondering if we can do something more specific around whatever edge case caused that.

Reply:
You can see the original issue here: https://github.com/Budibase/budibase/issues/6013 using the IDs of a query block in automations in a delete block loop.